Sources/App/Commands/Ingest.swift

```diff
@@ -124,13 +124,22 @@ func ingest(client: Client,
                         Current.logger().warning("storeS3Readme failed")
                         s3Readme = .error("\(error)")
                     }
+                    
+                    let fork: Fork?
+                    do {
+                        fork = try await getFork(on: database, metadata: metadata)
+                    } catch {
+                        fork = nil
+                        Current.logger().warning("updating forked from failed")
+                    }
```

Comment:
```suggestion
                    let fork = try? await getFork(on: database, metadata: metadata)
```

Let's just straight up assign it. `getFork` is technically throwing due to query but we could even write `try? await Package.query` there, to be honest.

In fact, I think

```swift
func getFork(on database: Database, metadata: Github.Metadata) async -> Fork? {
```

is better, because we're mixing using `nil` as an "error" value and `throws` at the same time - my bad!

Reply:
Also note that I'd left another little tweak to `getFork` in a follow up comment on the other PR https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/pull/3355#issuecomment-2330991491

So in total it should be

```swift
func getFork(on database: Database, parent: Github.Metadata.Parent?) async -> Fork? {
    guard let parentUrl = parent?.normalizedURL else { return nil }

    if let packageId = try? await Package.query(on: database)
        .filter(\.$url, .custom("ilike"), parentUrl)
        .first()?.id {
        return .parentId(packageId)
    } else {
        return .parentURL(parentUrl)
    }
}
```

Sorry for the back and forth!